lib/spack/spack/compilers/clang.py

```diff
@@ -180,15 +180,16 @@ def default_version(cls, comp):
             output = compiler('--version', output=str, error=str)
 
             ver = 'unknown'
-            match = re.search(r'^Apple LLVM version ([^ )]+)', output)
-            if match:
+            match = re.search(
                 # Apple's LLVM compiler has its own versions, so suffix them.
-                ver = match.group(1) + '-apple'
-            else:
+                r'^Apple LLVM version ([^ )]+)|'
                 # Normal clang compiler versions are left as-is
-                match = re.search(r'clang version ([^ )]+)', output)
-                if match:
-                    ver = match.group(1)
+                r'clang version ([^ )]+)-svn[~.\w\d-]*|'
+                r'clang version ([^ )]+)',
+                output
+            )
+            if match:
+                ver = match.group(match.lastindex)
```

Comment:
Two questions (I'm likely just missing something):

* It looks like we are now omitting `-apple` from the version string, is there a particular reason for this?
* The regex match only appears to have one group, so is there a reason you are using `match.lastindex`?

Reply:
> It looks like we are now omitting -apple from the version string, is there a particular reason for this?

No, I must have missed the suffix when changing the code. Will fix it.

> The regex match only appears to have one group, so is there a reason you are using match.lastindex?

To avoid a cascade of conditionals there's now a single regex with 3 groups. Each group represents a detected version and there could be only a single match at a time due to the regex.